bitnami/seaweedfs/templates/_helpers.tpl

```diff
@@ -0,0 +1,535 @@
+        {{- print (include "seaweedfs.mariadb.fullname" .) -}}
+    {{- end -}}
+{{- else -}}
+    {{- print .Values.externalDatabase.host -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the database port
+*/}}
+{{- define "seaweedfs.database.port" -}}
+{{- if .Values.mariadb.enabled }}
+    {{- print "3306" -}}
+{{- else -}}
+    {{- printf "%d" (.Values.externalDatabase.port | int ) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the database name
+*/}}
+{{- define "seaweedfs.database.name" -}}
+{{- if .Values.mariadb.enabled }}
+    {{- print .Values.mariadb.auth.database -}}
+{{- else -}}
+    {{- print .Values.externalDatabase.database -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the database user
+*/}}
+{{- define "seaweedfs.database.user" -}}
+{{- if .Values.mariadb.enabled }}
+    {{- print .Values.mariadb.auth.username -}}
+{{- else -}}
+    {{- print .Values.externalDatabase.user -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the database secret name
+*/}}
+{{- define "seaweedfs.database.secretName" -}}
+{{- if .Values.mariadb.enabled }}
+    {{- if .Values.mariadb.auth.existingSecret -}}
+        {{- print (tpl .Values.mariadb.auth.existingSecret .) -}}
+    {{- else -}}
+        {{- print (include "seaweedfs.mariadb.fullname" .) -}}
+    {{- end -}}
+{{- else if .Values.externalDatabase.existingSecret -}}
+    {{- print (tpl .Values.externalDatabase.existingSecret .) -}}
+{{- else -}}
+    {{- printf "%s-externaldb" (include "common.names.fullname" .) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns an init-container that waits for the database to be ready
+*/}}
+{{- define "seaweedfs.filer.waitForDBInitContainer" -}}
+- name: wait-for-db
+  image: {{ include "common.images.image" (dict "imageRoot" .Values.mariadb.image "global" .Values.global) }}
+  imagePullPolicy: {{ .Values.mariadb.image.pullPolicy }}
+  {{- if .Values.filer.containerSecurityContext.enabled }}
+  securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.filer.containerSecurityContext "context" $) | nindent 4 }}
+  {{- end }}
+  command:
+    - bash
+  args:
+    - -ec
+    - |
+      #!/bin/bash
+
+      set -o errexit
+      set -o nounset
+      set -o pipefail
+
+      . /opt/bitnami/scripts/liblog.sh
+      . /opt/bitnami/scripts/libvalidations.sh
+      . /opt/bitnami/scripts/libmariadb.sh
+      . /opt/bitnami/scripts/mariadb-env.sh
+
+      info "Waiting for host $DATABASE_HOST"
+      mariadb_is_ready() {
+          if ! echo "select 1" | mysql_remote_execute "$DATABASE_HOST" "$DATABASE_PORT_NUMBER" "$DATABASE_NAME" "$DATABASE_USER" "$DATABASE_PASSWORD"; then
+             return 1
+          fi
+          return 0
+      }
+      if ! retry_while "mariadb_is_ready"; then
+          error "Database not ready"
+          exit 1
+      fi
+      info "Database is ready"
+  env:
+    - name: BITNAMI_DEBUG
+      value: {{ ternary "true" "false" (or .Values.mariadb.image.debug .Values.diagnosticMode.enabled) | quote }}
```

Comment:
`mariadb.image` is not declared in `values.yaml` file

Reply:
Is it necessary? It's part of the sub-chart values. Please note the lines above...
```yaml
- name: wait-for-db
  image: {{ include "common.images.image" (dict "imageRoot" .Values.mariadb.image "global" .Values.global) }}
  imagePullPolicy: {{ .Values.mariadb.image.pullPolicy }}
```
... perfectly renders using the sub-chart values:
```console
$ helm template seaweedfs bitnami/seaweedfs -stemplates/filer/statefulset.yaml
(...)
        - name: wait-for-db
          image: registry-1.docker.io/bitnami/mariadb:11.2.2-debian-11-r6
          imagePullPolicy: IfNotPresent
```